src/coreComponents/common/format/table/TableLayout.hpp

```diff
@@ -51,68 +55,328 @@ class TableLayout
+
+    /**
+     * @brief Sets the values alignment for the column.
+     * @param headerAlignment The alignment to set for the column values (left, right, or center).
+     * @return The current column object
+     */
+    TableLayout::Column & setValuesAlignment( Alignment valueAlignment );
+
+    /**
+     * @brief Checks if the column has any child columns.
+     * @return bool True if the column has child columns, otherwise false.
+     */
+    bool hasChild() const;
+
+    /**
+     * @brief Checks if the column has a parent column.
+     * @return bool True if the column has a parent, otherwise false.
+     */
+    bool hasParent() const;
+
+private:
+    /// The maximum string size of the column.
+    size_t m_maxStringSize;
+  };
+
+  /**_est columns / sub columns.
+   * An exemple of an iteration: A -> B.A -> B.B -> B.C -> C.A.A -> C.A.B -> C.B.A -> C.B.B -> D
    */
-  struct Column
+  class LeafIterator
   {
-    /// Structure who contains parameters for a column
-    ColumnParam m_parameter;
-    /// A vector containing all column values
-    std::vector< string > m_columnValues;
-    /// The largest string in the column
-    string m_maxStringSize;
+public:
+    using ColumnType = Column;
+
+    LeafIterator( ColumnType * columnPtr, size_t idxLayer ):
+      m_currentColumn( columnPtr ), m_currentLayer( idxLayer )
+    {}
+
+
+    LeafIterator & operator=( Column * columnPtr )
+    {
+      this->m_currentColumn= columnPtr;
+      return *this;
+    }
+
+    // Prefix ++ overload
+    LeafIterator & operator++()
+    {
+      if( m_currentColumn->m_next != nullptr )
+      {
+        // chercher le dernier sous-enfant du suivant //todo
+        m_currentColumn = m_currentColumn->m_next;
+        while( m_currentColumn->hasChild() )
+        {
+          m_currentLayer++;
+          m_currentColumn = &m_currentColumn->m_subColumn[0];
+        }
+      }
+      else
+      {
+        bool const hasParent = (m_currentColumn->m_parent != nullptr);
+        m_currentLayer -= size_t( hasParent );
+        m_currentColumn = hasParent ? m_currentColumn->m_parent : nullptr;
+      }
+      return *this;
+    }
+
+    // Postfix ++ overload
+    LeafIterator operator++( int )
+    {
+      LeafIterator temp = *this;
+      ++(*this);
+      return temp;
+    }
+
+    ColumnType & operator*()
+    {
+      return *m_currentColumn;
+    }
+
+    ColumnType * operator->()
+    {
+      return m_currentColumn;
+    }
+
+    friend bool operator== ( LeafIterator const & a, LeafIterator const & b )
+    {
+      return a.m_currentColumn == b.m_currentColumn;
+    };
+    friend bool operator!= ( LeafIterator const & a, LeafIterator const & b )
+    {
+      return a.m_currentColumn != b.m_currentColumn;
+    };
+
+    size_t getCurrentLayer() const
+    {
+      return m_currentLayer;
+    }
+
+private:
+    ColumnType * m_currentColumn;
+    size_t m_currentLayer;
   };
 
+  LeafIterator beginLeaf()
```

Comment:
Please document the behaviour of the iteration.
```suggestion
  /**
   * @see LeafIterator
   */
  LeafIterator beginLeaf()
```
and also for the `end()`.